### DISCUSSION TITLE
Populating a tree with parameterised URL?

### DISCUSSION THREAD

**Discussion by Pete6666** [vakata/jstree#2546] [Q&A] [answered]
Would appreciate some help with this. The code below works fine and populates my tree perfectly. The php file returns all the data in json format from my mysql table which has the structure:

| id | name | parentid |

I now want to limit the data returned depending on the value in a field I have added to the table such that the table now has this structure: eg: I want to return the data that has userid = 'A'

| **userid** | id | name | parentid |

I have read the information in the jstree manual which appears to indicate some additional entries are required in the 'data' section of the config. But I am lost as to how to implement them. An example would be most welcome.

Thanks for any help.

    function buildstaffallocatertreeview() 
	{                       			
			$('#staffallocatertree').jstree('refresh');
			$('#staffallocatertree').jstree(
			{
                'plugins': ["json_data", "sort", "ui", 'checkbox'], 
				'core': 
				{              
			      "check_callback": true,
			     
                  'data' : 
				  {
					  "url" : "tl2_get_allstaff_as_tree.php",	                     					  
					  "dataType" : "json" 			
				  },	

                  'themes':
				  {
					'dots': false,
					'icons': false
				  },				 
                },
					                																							
				'checkbox': 	   	   	   
				{       
					'three_state'   : false,  				
					'tie_selection' : false,  					
				}							
            });			         		  
    }

**Comment by vakata** [comment, marked as answer]
URL can be a function (callback) which receives the node which the tree wants to load.

**Reply by Pete6666** [reply]
Ok, Thank you.
